tools/scripts/stm32.mk

```diff
@@ -0,0 +1,95 @@
+CFLAGS += -std=gnu11 \
+	-g3 \
+	-DUSE_HAL_DRIVER \
+	-DDEBUG \
+	-O0 \
+	-ffunction-sections \
+	-fdata-sections \
+	--specs=nano.specs \
+	-mfpu=fpv4-sp-d16 \
+	-mfloat-abi=hard \
+	-mthumb \
+	-mcpu=cortex-m4
+
+LDFLAGS	= -mcpu=cortex-m4 \
+	--specs=nosys.specs \
+	-Wl,--gc-sections \
+	-static \
+	--specs=nano.specs \
+	-mfpu=fpv4-sp-d16 \
+	-mfloat-abi=hard \
+	-mthumb \
+	-Wl,--start-group \
+	-lc \
+	-lm \
+	-Wl,--end-group
+
+CC = arm-none-eabi-gcc
+AS = arm-none-eabi-gcc
+AR = arm-none-eabi-ar
+
+rwildcard=$(foreach d,$(wildcard $(1:=/*)),$(call rwildcard,$d,$2) $(filter $(subst *,%,$2),$d))
+ifneq '' '$(call rwildcard,src,stm32f4*)'
+SUBPLATFORM = stm32f4
+CFLAGS += -I$(NO-OS)/libraries/stm32/STM32CubeF4/Drivers/STM32F4xx_HAL_Driver/Inc \
+	-I$(NO-OS)/libraries/stm32/STM32CubeF4/Drivers/STM32F4xx_HAL_Driver/Inc/Legacy \
+	-I$(NO-OS)/libraries/stm32/STM32CubeF4/Drivers/CMSIS/Device/ST/STM32F4xx/Include \
+	-I$(NO-OS)/libraries/stm32/STM32CubeF4/Drivers/CMSIS/Include
+TARGETCFG = target/stm32f4x.cfg
+else
+$(error Couldn't detect stm32 family.$(ENDL))
+endif
+
+openocd_paths:
+ifndef OPENOCD_SCRIPTS
+	$(error OPENOCD_SCRIPTS not found in shell environment.$(ENDL))
+endif
+
+ifndef OPENOCD_BIN
+	$(error OPENOCD_BIN not found in shell environment.$(ENDL))
+endif
+
+.PHONY: $(BINARY).openocd
```

Comment:
Use PHONEY += $(BINARY).openocd
At the end of generic.mk  .PHONEY = $(PHONEY)

Reply:
Anyway is there a reason to set this rules as PHONY ?
I think every time you call run will be overwritten with the same value.
If you don't set PHONY  then they will be written once.

Reply:
I mean, I don't get why maintain a PHONY list when it doesn't add any value ? You still have to add one line before each rule so you're not even saving typing.

I agree for the other comment, I need to see if they really need to be rebuilt all the time.

Reply:
Did a test regarding usefulness of those rules being PHONY: For example, the user may wanna edit the gdb script to uncomment some of the commented lines in stm32.mk or add new persistent gdb commands at startup. If the rule is not marked as .PHONY, it will not update the file build/project.elf.gdb.

Reply:
I took the PHONY approach from the makefile in linux OS, I don't know why :)
anyway you should use .PHONEY += something otherwise  with := will be overwritten.

Reply:
so conclusion here is I'm removing .PHONY: dependence on .openocd and .gdb rules